### PR TITLE
DEP: Alias hanning as an explanatory function of hann

### PR DESCRIPTION
**Pull request by endolith**
So there aren't duplicate help pages misleading people into thinking
that they're subtly different.

As in https://github.com/scipy/scipy/pull/228#issuecomment-343726453

### CODE REVIEW

**Comment by endolith** [comment]
I don't understand the build errors.

> Exception occurred:
  File "/home/ubuntu/scipy/build/testenv/lib/python2.7/site-packages/scipy/signal/__init__.py", line 345, in deco
    raise RuntimeError('dev error: badly formatted doc')
RuntimeError: dev error: badly formatted doc

:/

**Comment by pv** [comment]
Looking into the last file in the traceback probably clarifies the issue.

File: scipy/signal/windows/windows.py [hunk 1]
```diff
@@ -816,7 +816,13 @@ def hann(M, sym=True):
...
 
 
-hanning = hann
+def hanning(*args, **kwargs):
+    """
+    Alias of `hann`
+
+    This is for backwards compatibility. See `hann` for more details.
+    """
+    hann(*args, **kwargs)
```
Comment (WarrenWeckesser) [inline scipy/signal/windows/windows.py]: Missing **`return`** statement.  This line should be

    return hann(*args, **kwargs)

**Comment by WarrenWeckesser** [comment]
From the discussion in https://github.com/scipy/scipy/pull/228, it looks like `hanning` needs a complete docstring.  It was suggested there that some sort of decorator for aliases be created, but in the meantime, the docstring for `hanning` could be copied from `hann`, with an additional note added at the top stating that it is an alias for `hann`.

File: scipy/signal/windows/windows.py (hunk 1 above)
Comment (endolith) [reply scipy/signal/windows/windows.py]: derp. I'll fix that.

**Comment by endolith** [comment]
> the docstring for hanning could be copied from hann, with an additional note added at the top stating that it is an alias for hann.

Ok I'll do that for now.

**Comment by endolith** [comment]
(Or this could just wait until https://github.com/scipy/scipy/issues/8319)

**Comment by endolith** [comment]
I changed it to use numpy.deprecate instead:

```
@np.deprecate(new_name='hann')
def hanning(*args, **kwargs):
    return hann(*args, **kwargs)
```

Now it's being blocked by deco again.

@larsoner it looks like you create the deco function in https://github.com/scipy/scipy/pull/7900, do you have an opinion on it blocking these aliases?  It's probably a good idea to make PRs fail if function docstrings don't have *Parameters* sections, but in the case of aliases I think it's ok.

**Comment by pv** [comment]
You can just remove `hanning` from `deprecated_windows` since `np.deprecated` already adds the deprecation warning. The `new_name` however likely should be the full name, `scipy.signal.windows.hann`.

**Comment by endolith** [comment]
@pv ok I did that.

**Pull request merged**